app/validators/authorize_associations_validator.rb

```
@@ -0,0 +1,107 @@
+#
+# Copyright 2014 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+class AuthorizeAssociationsValidator < ActiveModel::Validator
+
+  def validate(record)
+    if record.class.respond_to?(:reflect_on_all_associations)
+      exemptions = if record.respond_to?(:association_authorization_overrides)
+                     record.association_authorization_overrides
+                   else
+                     {}
+                   end
+
+      record.class.reflect_on_all_associations.each do |association|
+
+        if check_authorization?(association, exemptions)
+
+          permission = exemptions[association.name] || "view_#{association.klass.name.downcase.pluralize}"
+          check_association(record, association, permission)
+
+        end
+      end
+    end
+  rescue NameError => e
```

This swallows exceptions and as a result grants access to associations that shouldn't be granted.